src/WebJobs.Script/Workers/Rpc/IWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc
+{
+    /// <summary>
+    /// Interface to resolve the Worker Configs.
+    /// </summary>
+    internal interface IWorkerConfigurationResolver
+    {
+        List<string> GetWorkerConfigPaths();
+
+        IOptionsMonitor<WorkerConfigurationResolverOptions> GetWorkerConfigurationResolverOptions();
```

Comment:
I wouldd suggest not exposing `WorkerConfigurationResolverOptions` through the interface. It feels like an internal implementation detail specific to `DefaultWorkerConfigurationResolver` implementation.

Instead, we could define an **immutable** type like this to encapsulate the relevant data:
```csharp
internal sealed record WorkerConfigurationInfo
{
    public required string BasePath { get; init; }

    public required IReadOnlyList<string> ConfigPaths { get; init; }
}
```
and update the interface to return that

```
internal interface IWorkerConfigurationResolver
{
    WorkerConfigurationInfo GetConfiguration();
}
```    

This keeps the interface clean and focused on what it provides, not how it is implemented.